presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -317,7 +317,7 @@ private void processNewSources()
                 }
 
                 // if current operator is finished...
-                if (current.isFinished()) {
+                if (current.isFinished() && !next.isFinished() && isBlocked(next).isDone()) {
```

Comment:
@dain, but many (some?) operators relays on `isBlocked()` preventing from calling `addInput()` and maybe other methods. For example `PartitionedOutputOperator` and `TaskOutputOperator` both have following code:
```
    @Override
    public void addInput(Page page)
    {
        requireNonNull(page, "page is null");
        checkState(isBlocked().isDone(), "output is already blocked");
        (...)
    }
```